### ISSUE TITLE
New tanks collider issues?

### ISSUE THREAD

**Issue by Pappystein** [CobaltWolf/Bluedog-Design-Bureau#424]
Both the LDC tanks and the Delta III topper tank seem to have collider issues.

Delta III tank seems to have a 1.5m diameter collider (the 2.5m area attaches as if it was 1.5m)  This is for both attaching things to the 2.5m section or attaching the tank AT the 2.5m section)

the LDC tanks seem to have a 3.75m collider for when they attach to things laterally (on the side skins)  But when something is attached to them they work fine....

**Comment by jsolson** [comment]
The collider looks is fine on the Delta III tank. The surface attach node was off. Same with the LDC tanks.

Incorrect or missing surface attach nodes are a chronic problem in this mod. If you throw them at me not too many at a time I'l fix them. Or anyone do it, they need:

~~~
node_attach = 1.25, 0.0, 0.0, 1.0, 0.0, 0.0, 1
~~~
where the 1.25 is the radius.

**Comment by CobaltWolf** [comment]
I'll try and remember to fix them when I throw new parts together.

**Comment by Pappystein** [comment]
How does that work for Non linear parts (Delta III for example being a part cylinder and part conic?

**Comment by CobaltWolf** [comment]
it depends on where you want them to attach. Usually the widest point.

**Comment by jsolson** [comment]
Pappystein the node_attach on the part being attached attaches to the collider (the surface) of the part it's being attached to. I can't find an example of a angled part that uses it so I don't know if you could change the angle of the node.

https://wiki.kerbalspaceprogram.com/wiki/CFG_File_Documentation#Node_Definitions
node_attach = x, y, z, angx, angy, angz, size

node_stack_top and bottom use 1.0 and -1.0 for angy, node attach uses 1.0 in either angx (and maybe angy). I don't know if 0.5 would work for 45 degrees (or maybe sin(45) - I dunno). You might explode the universe.

**Comment by jmrd98** [comment]
I've been down this road before. The wiki docs for attach nodes are sparse to be sure. There are three references I would see on the topic: [This thread from the modeling subforum](https://forum.kerbalspaceprogram.com/index.php?/topic/136106-rotating-attachment-nodes/), the [NodeHelper mod](https://forum.kerbalspaceprogram.com/index.php?/topic/115700-140-nodehelper-tool-used-to-edit-node-positions/&) and [the universal storage MM for the Apollo SM I wrote forever ago.](https://github.com/CobaltWolf/Bluedog-Design-Bureau/blob/master/Gamedata/Bluedog_DB/Compatibility/UniversalStorage/Apollo_Block2_ServiceModule_UniversalStorage.cfg)

tldr: 45 degree angles are easy for defined nodes because trig, but other angles are a slight pain because trig. 

But, i'm with jso on this one: I don't know what the results would be for the surface attach type. If the angle is constant, or near constant, I could see defining a surface node as the normal vector to the 'flat side', but I haven't seen these parts in game so I am not sure if that would work here.

**Comment by jsolson** [comment]
You're right, they would be vectors. I hate vectors

**Comment by CobaltWolf** [comment]
The good thing is we never need to do anything other than 0, 1, -1 for those ;)
